components/script/dom/indexeddb/idbtransaction.rs

```diff
@@ -49,11 +50,25 @@ pub struct IDBTransaction {
     active: Cell<bool>,
     // https://www.w3.org/TR/IndexedDB-2/#transaction-finish
     finished: Cell<bool>,
+    // https://www.w3.org/TR/IndexedDB-2/#transaction-commit
+    commit_requested: Cell<bool>,
+    // https://www.w3.org/TR/IndexedDB-2/#transaction-commit
+    committing: Cell<bool>,
     // Tracks how many IDBRequest instances are still pending for this
     // transaction. The value is incremented when a request is added to the
     // transaction’s request list and decremented once the request has
     // finished.
     pending_request_count: Cell<usize>,
+    aborted: Cell<bool>,
+    upgrade_old_version: Cell<Option<u64>>,
+    // “These steps are invoked by [HTML]… ensure that transactions created by a script call to
+    // transaction() are deactivated once the task that invoked the script has completed. The steps
+    // are run at most once for each transaction.”
+    // https://w3c.github.io/IndexedDB/#cleanup-indexed-database-transactions
+    #[no_trace]
+    cleanup_event_loop: DomRefCell<Option<ScriptEventLoopId>>,
```

Comment:
So I don't think we need the `ScriptEventLoopId`, and also we shouldn't store this on the transaction. Instead we should store transactions on the global, and then clean-up transactions for the global at the microtask checkoint: see step 5 of https://html.spec.whatwg.org/#perform-a-microtask-checkpoint.